api_products.md

```diff
@@ -11,25 +11,214 @@ information about api access.
+
+#### Supported Units
+
+The supported units are:
+
+| Abbreviation | Description                                                                                                    |
+|--------------|----------------------------------------------------------------------------------------------------------------|
+| ml           | Mililiter                                                                                                      |
+| dl           | Deciliter                                                                                                      |
+| l            | Liter                                                                                                          |
+| m3           | Cubic meter                                                                                                    |
+| cm3          | Cubic centimeter                                                                                               |
+| m2           | Square meter                                                                                                   |
+| cm2          | Square centimeter                                                                                              |
+| mm           | Millimeter                                                                                                     |
+| cm           | Centimeter                                                                                                     |
+| m            | Meter                                                                                                          |
+| kg           | Kilogram                                                                                                       |
+| g            | Gram                                                                                                           |
+| t            | Tonne                                                                                                          |
+| piece        | The encoded number should be interpreted as number of contained units, i.e. number of bread rolls in a package |
+| price        | The encoded number should be interpreted as total price, i.e. price of a package of different sausages         |
+
+
+### Code Template
+
+Code templates are a way to extract information embeded into a scanned
+code. For example a instore EAN-13 has the format `2xxxxxCeeeeeC`
+where the five `x` encode an identfier for the product, the five `e`
+encode the the information and the `C` are the checksums defined by
+the standard. With this and the additional inforations contained in
+the product, it is possible to calculate the price.
+
+For example: Given the scanned code `2123455005005`. Using the
+template it is possible to extract the identifier `12345` and the
+embeded information `00500`. A lookup in products information might
+yield, that this code belongs to the product `Banana` wich has a price
```

Comment:
A product lookup might yield, that this code belongs to the product `Banana` which has a price